static/js/subs.js

```diff
@@ -915,6 +915,16 @@ exports.change_stream_name = function (e) {
 
     $("#subscriptions-status").hide();
 
+    // While renaming we need to check if the name is free or has already been taken up.
+    var stream_status = compose.check_stream_existence(new_name);
+
+    if (stream_status !== "does-not-exist") {
+        $("#name_change_error_" + stream_id).text(i18n.t("A stream with this name already exists!"));
+        $("#name_change_error_" + stream_id).show();
+    } else {
+        $("#name_change_error_" + stream_id).hide();
+    }
```

Comment:
(Also, we're trying to eliminate `compose.check_stream_existence` anyway, since it's a bad function)